cleanlab_pinecone_rag_workflow/Cleanlab_Pinecone_RAG.ipynb

```diff
@@ -0,0 +1,5263 @@
+    "        Returns:\n",
+    "            pd.DataFrame: DataFrame containing chunk data and metadata.\n",
+    "                          Columns include all metadata fields stored in the index.\n",
+    "\n",
+    "        Raises:\n",
+    "            Exception: If there's an error retrieving chunks from the index.\n",
+    "        \"\"\"\n",
+    "        stats = self.index.describe_index_stats()\n",
+    "        total_vectors = stats.total_vector_count\n",
+    "        dimension = stats.dimension\n",
+    "\n",
+    "        print(f\"Index name: {self.index_name}\")\n",
+    "        print(f\"Total vectors according to stats: {total_vectors}\")\n",
+    "        print(f\"Vector dimension: {dimension}\")\n",
+    "\n",
+    "        try:\n",
+    "            results = self.index.query(\n",
+    "                vector=[0.0] * dimension,\n",
+    "                top_k=max_chunks,\n",
+    "                include_values=False,\n",
+    "                include_metadata=True,\n",
+    "            )\n",
+    "\n",
+    "            chunk_data = [match.metadata for match in results.matches]\n",
+    "            chunk_df = pd.DataFrame(chunk_data)\n",
+    "\n",
+    "            print(f\"Total chunks retrieved: {len(chunk_df)}\")\n",
+    "            return chunk_df\n",
+    "\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error retrieving chunks from index: {e}\")\n",
+    "            return pd.DataFrame()\n",
+    "\n",
+    "    def delete_chunks(self, chunk_ids: List[str]) -> None:\n",
+    "        \"\"\"\n",
+    "        Delete specific chunks from the Pinecone index.\n",
+    "\n",
+    "        Args:\n",
+    "            chunk_ids (List[str]): List of chunk IDs to delete.\n",
+    "\n",
+    "        Raises:\n",
+    "            Exception: If there's an error during the deletion process.\n",
+    "        \"\"\"\n",
+    "        try:\n",
+    "            self.index.delete(ids=chunk_ids)\n",
+    "            print(f\"Successfully deleted {len(chunk_ids)} chunks from the index.\")\n",
+    "        except Exception as e:\n",
+    "            print(f\"Error deleting chunks: {e}\")\n",
+    "            print(f\"Problematic chunk IDs: {chunk_ids}\")\n",
+    "\n",
+    "        print(f\"Finished deletion process for {len(chunk_ids)} chunks.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "04890aea",
+   "metadata": {},
+   "source": [
+    "Let's use the RAG pipeline we have defined above to create a Pinecone index - then we upsert our document chunks into our vector DB in batches."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "076081ad",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating new index: cleanlab-pinecone-index\n",
+      "Starting document processing...\n",
+      "Processing document documents/Blackstone-Third-Quarter-2023-Investor-Call.pdf at index 0...\n",
```

Comment:
The output of this cell is way too long for the notebook, please clear this cell's output